pkg/hostagent/setup.go

```diff
@@ -112,6 +114,27 @@ func (*ClientRPC) SetupVeth(args *SetupVethArgs, result *SetupVethResult) error
 
 		result.HostVethName = hostVeth.Name
 
+		// Force a consistent MAC address based on the IPv4 address
+		// Currently we dont have a support for V6.
+		// This code has to be revisted if upstream drops SetHWAddrByIP in future
+		//https://github.com/containernetworking/plugins/blob/e1517e2498fe4774435bc4be6bdd39fa735b469b/pkg/ip/link_linux.go#L228
+
+		switch {
+		case args.Ip.To4() == nil && args.Ip.To16() == nil:
+			return fmt.Errorf("neither ip4 or ip6 specified: %v", err)
+
+		case args.Ip.To4() != nil:
+			{
+				if err := ip.SetHWAddrByIP(args.IfName, args.Ip, nil); err != nil {
+					return fmt.Errorf("failed to set pod interface MAC address for v4: %v", err)
+				}
+			}
+		case args.Ip.To16() != nil:
+			{
+				//TO DO. Upstream doesn't handle ipv6 based mac allocation yet
+			}
+		}
```

Comment:
The above switch logic is also present in the SetHWAddrByIP implementation. Perhaps we could just check here if its an ipv4 and then call SetHWAddrByIP? (This is a minor comment, not required to merge this PR, we can do this as a follow-up patch too.)

Reply:
That's how the previous implementation looked like where I just checked for ipv4 and called the func SetHWAddrByIP. Since @jayaramsatya had a request to handle ipv6 , I changed the logic to the above one. I can  revert the changes if you insist.

Reply:
Just confirmed with Jayaram. The ask was to handle ipv6 by printing a log instead of siliently ignoring.  I will revert the changes and will switch to If condition based logic